vernac/vernacentries.mli

```diff
@@ -44,6 +44,23 @@ val interp_redexp_hook : (Environ.env -> Evd.evar_map -> Genredexpr.raw_red_expr
 (** Helper *)
 val vernac_require_open_proof : pstate:Proof_global.t option -> (pstate:Proof_global.t -> 'a) -> 'a
 
+val with_pstate : pstate:Proof_global.t option -> (pstate:Proof_global.pstate -> 'a) -> 'a
+
+val modify_pstate : pstate:Proof_global.t option -> (pstate:Proof_global.pstate -> Proof_global.pstate) -> Proof_global.t option
+
 (* Flag set when the test-suite is called. Its only effect to display
    verbose information for `Fail` *)
 val test_mode : bool ref
+
+(** For mlg *)
+
+type functional_vernac =
+  | VtDefault of (unit -> unit)
+  | VtModifyProofStack of (pstate:Proof_global.t option -> Proof_global.t option)
+  | VtMaybeOpenProof of (is_nested:bool -> Proof_global.pstate option)
+  | VtOpenProof of (is_nested:bool -> Proof_global.pstate)
+  | VtModifyProof of (pstate:Proof_global.pstate -> Proof_global.pstate)
+  | VtReadProofOpt of (pstate:Proof_global.pstate option -> unit)
+  | VtReadProof of (pstate:Proof_global.pstate -> unit)
+
+val interp_functional_vernac : functional_vernac -> st:Vernacstate.t -> Vernacstate.t
```

Comment:
This type is quite misleading and will easily induce to errors, as the interpretation is not functional on `Vernacstate.t` but only on `Proof_global.Stack.t`

Please update.